### DISCUSSION TITLE
Why is eth_signTransaction not supported?

### DISCUSSION THREAD

**Discussion by rubydusa** [NomicFoundation/hardhat#2271] [Troubleshooting]
(I'm using hardhat ethers, I don't know if this issue is universal across different hardhat plugins or not)

I'm writing a lot of unitests and I thought that it would be more convenient to have one account that would send all transactions and all the other accounts would only sign them, but I found out eth_signTransaction is not supported, why?

I feel like signing transactions is a relatively standard feature in all sorts of blockchain applications and because of that I am confused to why wouldn't such a feature be implemented in hardhat, especially considering that internally hardhat uses ethereumjs/vm that can support this RPC call

at the very least, hardhat should provide some low level way to interact with underlying hardhat network, so people can modify the hardhat network to their needs. I read the hardhat network section on the docs website a couple of times now and I haven't seen anything that might help changing the settings of the underlying test network

alternatively, some other way to create meta-transactions is helpful, but from my own expirience libraries and plugins relating to that sort of stuff tend to be... very opinionated...

Interested to hear what other people think about this, if anyone knows how to sign transactions in hardhat that'll be very helpful - I've been digging around for hours with no hope

**Comment by fvictorio** [comment]
Hi @rubydusa. The original reason why this method wasn't supported is that is not widely supported by wallets and clients. For example, Metamask doesn't support it. I think `geth` does, but other clients don't (I might be wrong about this though).

If you want to sign txs like that, you can create `Wallet` objects with ethers using the private keys, and sign them. Right now Hardhat doesn't have a great way of obtaining the PKs in your config, but it's doable.

That being said, I would like to understand your use case better. Especially this: " I thought that it would be more convenient to have one account that would send all transactions and all the other accounts would only sign them". Why do you find that more convenient?
